starter_code/public/javascript/APIHandler.js

```diff
@@ -4,22 +4,59 @@ class APIHandler {
   }
 
   getFullList () {
-
+    return axios.get("http://localhost:8000/characters")
+        .then(response => {
+          console.log(response.data);
+          return response.data;
+        })
+        .catch(err => {
+          console.log("ERROR", err);
+        });
   }
 
-  getOneRegister () {
-
+  getOneRegister (id) {
+     return axios.get("http://localhost:8000/characters/" + id)
+    .then(response => {
+      console.log(response.data);
+      return response.data;
+    })
+    .catch(err => {
+      console.log("ERROR", err);
+    });
   }
 
-  createOneRegister () {
-
+  createOneRegister(elements) {
+    return axios.post("http://localhost:8000/characters", elements)
+    .then(response => {
+      console.log(response.data);
+      return response.data;
+    })
+    .catch(err => {
+      console.log("ERROR", err);
+    });
   }
 
-  updateOneRegister () {
-
+  updateOneRegister (id, update) {
+    return axios.put("http://localhost:8000/characters/" + id, update)
```

Comment:
You might want to use a `patch` verb here instead of `put`, as `put` will update the whole file, not just one.